### PR TITLE
#955 stepprocess: feature to call selectStep when isSelected is set to true on one of the SohoStepListItemComponent

### LINKED ISSUE #955: Step process: when setting isSelected to true should select that panel into view
**Is your feature request related to a problem? Please describe.**
We navigate to another panel via code in the stepprocess component in enterprise-ng, we can only navigate to a panel via these three actions:
- clicking next
- clicking previous
- selecting the panel from the step list

**Describe the solution you'd like**
Created a pull request when we set a SohoStepListItemComponent isSelected property to true, it will call the selectStep and should navigate to that panel and be shown into view.

**Describe alternatives you've considered**
None yet

**Additional context**
https://jira.lawson.com/browse/LMCLIENT-32916

### PR DESCRIPTION
**Pull request by JBismonte**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

We cannot navigate to another panel via code, in this PR, we can just update the stepListItemComponent of that particular panel to true, and it will call selectStep to navigate and display that particular panel into view. 

**Related github/jira issue (required)**:

- Closes #955 
- https://jira.lawson.com/browse/LMCLIENT-32916

**Steps necessary to review your pull request (required)**:

- pull the code from a fork: https://github.com/JBismonte/enterprise-ng, build and run the demo app.
- go to http://localhost:4200/ids-enterprise-ng-demo/step-process-auto-step-change
- click any buttons, it should navigate to the correct panel.

See screen recording:

https://user-images.githubusercontent.com/29673378/102778777-25b6e800-43ce-11eb-8334-5a5fadb60231.mp4

note: 
- when you set the isSelected to true in a panel that is a folder, it will toggle the folder to either open or close.

### CODE REVIEW

**Review by tmcconechy — DISMISSED** [review]

**Review by nbcp — CHANGES_REQUESTED** [review]
@JBismonte please also include your changes for the demo page here. That way it can be used for testing this fix and future use cases.

File: src/app/stepprocess/stepprocess-auto-step-change.demo.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    setTimeout(() => {
+      if (this.stepListItems) {
+        const stepListItemsLength = this.stepListItems.toArray().length;
+        const stepNo = Math.floor(Math.random() * stepListItemsLength);
+        this.stepListItems.toArray()[stepNo].isSelected = true;
+        console.log(stepNo);
+        console.log('step changed');
+      }
+    }, 11000);
+
```
Comment (nbcp) [inline src/app/stepprocess/stepprocess-auto-step-change.demo.ts:99]: lets make it just select one defined random sub step, this way we can add an automated test for it in the future.

Comment (nbcp) [reply src/app/stepprocess/stepprocess-auto-step-change.demo.ts:99]: Actually there's no need for this and the ViewChildren since isSelected is an Input in SohoStepListItemComponent, we can just add `[isSelected]="true"`  to one of the `soho-step-list-item` in the template

**Review by nbcp — CHANGES_REQUESTED** [review]

File: src/app/stepprocess/stepprocess-auto-step-change.demo.ts (hunk 1 above)
Comment (nbcp) [reply src/app/stepprocess/stepprocess-auto-step-change.demo.ts:99]: I just realize http://localhost:4200/ids-enterprise-ng-demo/step-process already does this so maybe we can just use that

Comment (JBismonte) [reply src/app/stepprocess/stepprocess-auto-step-change.demo.ts]: Hi @nbcp, tried to refactor the code of the demo page, and it will rely on the clicking of the buttons rather than awaiting the setTimeout functions in the ngAfterViewInit.

File: projects/ids-enterprise-ng/src/lib/stepprocess/soho-stepprocess.component.ts [hunk 2]
```diff
@@ -82,6 +82,8 @@ export class SohoStepListItemComponent {
     }
   }
 
+  @Input() stepId?: string;
```
Comment (nbcp) [inline projects/ids-enterprise-ng/src/lib/stepprocess/soho-stepprocess.component.ts:85]: why was this added? please remove it if it is only needed for the demo component

File: src/app/stepprocess/stepprocess-auto-step-change.demo.html [hunk 3]
```diff
@@ -8,21 +8,21 @@
 
     <ng-template ngFor let-step [ngForOf]="steps">
 
-      <li soho-step-list-item *ngIf="step.content">
+      <li soho-step-list-item stepId="{{step.id}}" *ngIf="step.content">
```
Comment (nbcp) [inline src/app/stepprocess/stepprocess-auto-step-change.demo.html:11]: why go adding a new input when you could just do something like:
`[isSelected]={{step.id == selectedStepId}}`

File: src/app/stepprocess/stepprocess-auto-step-change.demo.ts [hunk 4]
```diff
@@ -36,70 +38,23 @@ export class StepProcessAutoStepChangeComponent implements OnInit, AfterViewInit
...
-        console.log('step changed');
-      }
-    }, 11000);
 
+  selectPanel(id: string) {
+    console.log(id);
+    const selectedItem = this.stepListItems.find((item) => item.stepId === id);
+    if (selectedItem) {
+      selectedItem.isSelected = true;
+    }
```
Comment (nbcp) [inline src/app/stepprocess/stepprocess-auto-step-change.demo.ts:52]: and here your could just:
```suggestion
    this.selectedStepId = id;
```

**Review by nbcp — CHANGES_REQUESTED** [review]

File: projects/ids-enterprise-ng/src/lib/stepprocess/soho-stepprocess.component.ts [hunk 5]
```diff
@@ -73,7 +74,17 @@ export class SohoStepListItemComponent {
...
     return true;
   }
-  @HostBinding('class.is-selected') @Input() isSelected = false;
+
+  @HostBinding('class.is-selected') @Input() set isSelected(isSelected: boolean) {
+    if (isSelected) {
+      const stepLink = $(this.element.nativeElement).children('a.js-step-link');
+      this.sohoStepProcessComponent.stepprocess?.selectStep(stepLink);
+    }
+  }
```
Comment (nbcp) [inline projects/ids-enterprise-ng/src/lib/stepprocess/soho-stepprocess.component.ts:83]: Since this becomes a setter, we'll also need a to set the value in a private property and expose it via a getter. Hostbinding for class should be moved there as well

Comment (bthharper) [reply projects/ids-enterprise-ng/src/lib/stepprocess/soho-stepprocess.component.ts:83]: Agree with the above comment.

```typescript 

  @HostBinding('class.is-selected') _isSelected: boolean = false;

  @Input() set isSelected(isSelected: boolean) {
    this._isSelected = isSelected;
    if (isSelected) {
      const stepLink = $(this.element.nativeElement).children('a.js-step-link');
      this.sohoStepProcessComponent.stepprocess?.selectStep(stepLink);
    }
  }
```

**Review by nbcp — APPROVED** [review]
